airbyte-integrations/connectors/source-maestro-qa/source_maestro_qa/source.py

```diff
@@ -0,0 +1,274 @@
+        return {
+            **body,
+            "singleFileExport": self.report_type,
+            "includeCalibrations": "all",
+            "includeGraderQA": True,
+        }
+
+
+class Annotations(RawExport):
+    primary_key = ["gradable_id", "annotation_id", "date_reported_as"]
+    report_type = "annotations"
+
+
+class CSAT(RawExport):
+    report_type = "csat"
+
+
+class IndividualAnswers(RawExport):
+    primary_key = ["gradable_id", "question_id", "date_reported_as"]
+    report_type = "individual_answers"
+
+
+class SectionScores(RawExport):
+    primary_key = ["gradable_id", "section_id", "date_reported_as"]
+    report_type = "section_scores"
+
+
+class TotalScores(RawExport):
+    primary_key = ["gradable_id", "date_reported_as"]
+    report_type = "total_scores"
+
+
+class AuditLogs(IncrementalMaestroQAStream):
+    primary_key = ["activity_id", "time_stamp", "description"]
+
+    def path(self, **kwargs) -> str:
+        return "request-audit-log-export"
+
+
+class ScreenCaptureUsageStats(MaestroQAStream):
+    primary_key = "agentId"
+    _group_ids = []
+
+    def __init__(self, parent: MaestroQAStream, **kwargs):
+        self.parent = parent
+        super().__init__(**kwargs)
+
+    def request_body_json(self, **kwargs) -> Optional[Mapping]:
+        return {
+            "groupIds": self._group_ids,
+            "interval": "day",
+        }
+
+    def path(self, **kwargs) -> str:
+        return "screen-capture-usage-stats"
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        self._group_ids = list(
+            set([rec.get("group_id") for rec in self.parent.read_records(sync_mode, cursor_field, stream_slice, stream_state)])
+        )
+        return super().read_records(sync_mode, cursor_field, stream_slice, stream_state)
+
+
+class TokenAuth(TokenAuthenticator):
+    """Overwrite to allow token without auth method"""
+
+    @property
+    def token(self) -> str:
+        return self._token
+
+
+class SourceMaestroQA(AbstractSource):
+    def get_auth(self, config) -> TokenAuth:
+        return TokenAuth(token=config.get("api_key"), auth_method="", auth_header="apitoken")
+
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        try:
+            auth = self.get_auth(config)
+            stream = AgentGroups(authenticator=auth)
+            next(stream.read_records(sync_mode=SyncMode.full_refresh))
+            return True, None
+        except Exception as error:
+            return False, f"Unable to connect to MaestroQA API with the provided credentials - {repr(error)}"
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        auth = self.get_auth(config)
+        default_start_date = config.get("start_date")
+
+        agent_groups = AgentGroups(authenticator=auth)
```

Comment:
Why not move AgentGroups to the list?

Reply:
It is in the list but I'm also using it as a parent stream for `ScreenCaptureUsageStats`